azurerm/azurerm_kubernetes_cluster/simple/main.tf

```diff
@@ -0,0 +1,53 @@
+# Summary: A simple Azure Kubernetes Cluster
+# Explanation: Also known as Azure Kubernetes Service (AKS)
+
+# Documentation: https://www.terraform.io/docs/language/settings/index.html
+terraform {
+  required_version = ">= 0.14.0"
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "~> 2.0"
+    }
+  }
+}
+
+# Documentation: https://www.terraform.io/docs/language/values/variables.html
+variable "azure_subscription_id" {
+  type = string
+}
+
+# Documentation: https://www.terraform.io/docs/language/providers/requirements.html
+# Documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs
+provider "azurerm" {
+  features {}
+
+  subscription_id = var.azure_subscription_id
+}
+
+# Resource Group
+# Documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_group
+resource "azurerm_resource_group" "changeme_simple_k8s_rg" {
+  name     = "changeme-simple-k8s-rg"
+  location = "West Europe"
+}
+
+# Kubernetes Cluster within the Resource Group
+# Documentation: https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster
+resource "azurerm_kubernetes_cluster" "changeme_simple_k8s" {
+  name                = "changeme-simple-k8s"
+  location            = azurerm_resource_group.changeme_simple_k8s_rg.location
+  resource_group_name = azurerm_resource_group.changeme_simple_k8s_rg.name
+
+  dns_prefix = "changeme-simple-k8s-dns-prefix"
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  default_node_pool {
+    name       = "changeme"
+    node_count = 1
+    vm_size    = "standard_b1ls"
+  }
+}
```

Comment:
Another issue when running `terraform apply`:
* https://github.com/ContainerSolutions/terraform-examples/runs/3261241014?check_suite_focus=true#step:5:432

```
azurerm_kubernetes_cluster.changeme_simple_k8s: Still creating... [5m40s elapsed]
╷
│ Error: waiting for creation of Managed Kubernetes Cluster "changeme-simple-k8s" (Resource Group "changeme-simple-k8s-rg"): Code="CreateVMSSAgentPoolFailed" Message="AKS encountered an internal error while attempting the requested Creating operation. AKS will continuously retry the requested operation until successful or a retry timeout is hit. Check back to see if the operation requires resubmission. Correlation ID: 1ee4eb8a-23d8-cdcf-98be-de55bc67a879, Operation ID: 57d1af14-9ead-4b4a-9e57-f85ca9410ed9, Timestamp: 2021-08-06T10:27:35Z."
│ 
│   with azurerm_kubernetes_cluster.changeme_simple_k8s,
│   on main.tf line 37, in resource "azurerm_kubernetes_cluster" "changeme_simple_k8s":
│   37: resource "azurerm_kubernetes_cluster" "changeme_simple_k8s" ***
│ 
╵
/home/runner/work/terraform-examples/terraform-examples/azurerm/azurerm_kubernetes_cluster/simple
```

Reply:
Probably related to this:
* https://docs.microsoft.com/en-us/azure/aks/quotas-skus-regions#restricted-vm-sizes
```
Restricted VM sizes
Each node in an AKS cluster contains a fixed amount of compute resources such as vCPU and memory. If an AKS node contains insufficient compute resources, pods might fail to run correctly. To ensure the required kube-system pods and your applications can be reliably scheduled, don't use the following VM SKUs in AKS:

Standard_A0
Standard_A1
Standard_A1_v2
Standard_B1ls
Standard_B1s
Standard_B1ms
Standard_F1
Standard_F1s
Standard_A2
Standard_D1
Standard_D1_v2
Standard_DS1
Standard_DS1_v2
```

Reply:
Added a fix for this: https://github.com/ContainerSolutions/terraform-examples/pull/73/commits/cee640d775c36f9231c75987f0b940d1a41e1d42